Makefile

```diff
@@ -1,41 +0,0 @@
-include .env
```

Comment:
if you plan to remove totally the Makefile, it's not clear why, consider to update the CONTRIBUTING.md section.

IMHO Makefile are a really good starting point to create a set of commands for the contributors/developers - consider that when removing it, and create something equivalent ;)

Reply:
I'll update the docs, including `CONTRIBUTING.md` once the dust settles and I get all the workflows, releasing, etc. working. We're removing the Makefile because it's duplicative in functionality with `hatch` scripts.

We're moving to `hatch` across all adapters. We started with the `dbt-adapters` repo and the `dbt-postgres` repo since they were new. This felt like the right point here as well since we need to rewire some of the workflows for things like releasing. If you're unfamiliar with the tool, `hatch` feels a bit like a combination of `Makefile` and `tox`. We use a single default environment which is equivalent to `dev-requirements.txt` and we have scripts in there that are close to 1:1s with the `Makefile` recipes (which I can include in the docs). e.g.:
`make setup` -> `hatch run setup`
`make unit_test` -> `hatch run unit-tests`
`make functional_test` -> `hatch run integration-tests`
`make pre-commit-all` -> `hatch run code-quality`
Additionally, getting into the virtual environment is just `hatch shell`, similar to `python -m venv venv && source venv/bin/activate`.

Reply:
If you're interested, please take a look at the updated `CONTRIBUTING.md` [file](https://github.com/dbt-labs/dbt-athena/blob/actions/public-pypi-deploy/CONTRIBUTING.md) and let me know if this covers your concerns. I tried to capture each of the previous topics, but with the new tooling.

Reply:
First of all, thanks for the context appreciate the exhaustive answer - I've been a big fan of Makefiles, and I was not aware of `hatch` - pretty cool. I had a look at the `CONTRIBUTING.md` and looks good to me. Thanks for addressing my concerns. Feel free to mark this as resolved, I know that your PR is in a draft state, I just wanted to raise my concerns:)